### PR TITLE
[NHP-5] Implement main project

### PR DESCRIPTION
**Pull request by exxocism**
## JIRA issue/Description
- JIRA: https://exxocism.atlassian.net/browse/NHP-5
- This PR implements the main logic

## Major changes
- Main feature is implemented: [25e1002](https://github.com/exxocism/node-healthz-prober/pull/2/commits/25e1002524805e4a73782bbc961e4192f5042638), [3b3b938](https://github.com/exxocism/node-healthz-prober/pull/2/commits/3b3b938a8c1224f6c8a3951acbcc7328d66a2d2d), [9a0974f](https://github.com/exxocism/node-healthz-prober/pull/2/commits/9a0974f40a37d3368bb38af9d902472db1a27a69), [814b478](https://github.com/exxocism/node-healthz-prober/pull/2/commits/814b478a5270604e89e6e33d7e63ae3cb1db04ab)

## Minor changes
- Types added for slack reporting: [0f5cfbb](https://github.com/exxocism/node-healthz-prober/pull/2/commits/0f5cfbb6fe6963fbf96c5e87905f48d4e83264bf)
- Helping `.env.example` is added: [9066224](https://github.com/exxocism/node-healthz-prober/pull/2/commits/906622412f250a09d6bf842479eaa087a64b38a6)

## Screenshots
- To-be (success)
  001 prober
- To-be (failed)
  002 error

### 1. local development
- Install `node.js` by nvm: [windows](https://github.com/coreybutler/nvm-windows), [linux/mac](https://github.com/nvm-sh/nvm)
- Use node `20.10.0`: `nvm use`
- Install the required files: `npm ci`
### 4. How to test
- [ ] Check if `npm start` works properly
  ```
  Hello, world!
  ```
- [ ] Open .env and put some configurations with it: refer to `.env.example`
  https://github.com/exxocism/node-healthz-prober/blob/906622412f250a09d6bf842479eaa087a64b38a6/.env.example#L2-L7
- [ ] Start the application: `npm start`
- [ ] Check if the application probes the endpoints and reports to slack if it's not working
- [ ] Run `npm run runtime` and check if it builds the executable successfully
  - Linux/Mac `dist/executable`, Windows: `dist/executable.exe`
- [ ] Check if all the commands work in the different platform
  - [ ] Windows
  - [ ] Linux
  - [ ] MacOS (To be implemented)

## TODO
- Pre-deployment steps: None
- Post-deployment steps: None

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1 +1,16 @@
...
+    index++;
+    const { url, webhook } = getConfig(index);
+    if (!url) break;
+
+    console.log(`[${index}] Checking ${url}...`);
+    healthCheck(url, webhook);
+  }
+};
+
+mainFunction();
```
Comment (lasagnaslave) [inline src/index.ts:16]: (minor, non-blocker) How_ about indicating an app is running?

File: src/lib/getConfig.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { SlackChatData, SlackWebhook } from '../types/slack';
+
+export const getConfig = (index: number) => {
+  const url = process.env[`VITE_CHECK_URL_${index}`];
```
Comment (lasagnaslave) [inline src/lib/getConfig.ts:4]: Looks like the `VITE_` prefix could be irritating. can we make a utility function for this?

File: src/utils/slackUtils.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+export const sendToSlack = ({ webhookEndpoint, data }: SlackWebhook) => {
+  const payload = {
+    method: 'POST',
+    body: JSON.stringify(data),
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  };
+  fetch(webhookEndpoint, payload);
```
Comment (lasagnaslave) [inline src/utils/slackUtils.ts:11]: non-blocker) How about returning the value so we can handle it further?

**Review by lasagnaslave — APPROVED** [review]
Thanks for the effort!
LGTM with non-blocker comments 👍

File: src/index.ts (hunk 1 above)
Comment (exxocism) [reply src/index.ts:17]: Thanks for the suggestion! Applied in [2630738](https://github.com/exxocism/node-healthz-prober/pull/2/commits/263073883d01cfca3cf7d7660fd369a08093e725)

File: src/lib/getConfig.ts (hunk 2 above)
Comment (exxocism) [reply src/lib/getConfig.ts:4]: I think it increases the code complexity: If we suppose that we need a environmental variable without `VITE_ ` prefix. 🤔

File: src/utils/slackUtils.ts (hunk 3 above)
Comment (exxocism) [reply src/utils/slackUtils.ts]: Thanks for the suggestion! Applied in [7ff0f58](https://github.com/exxocism/node-healthz-prober/pull/2/commits/7ff0f5827a4da169267226c56fa7e7c748e6a75b)

**Review by lasagnaslave — APPROVED** [review]

**Review by lasagnaslave — COMMENTED** [review]
LGTM again 👍

**Pull request merged**